doc/source/tutorial/stats/rv_infrastructure.md

```diff
@@ -0,0 +1,920 @@
+```
+
+Compounding with these issues is that `fit` estimates all distribution parameters by default. In the example above, we are probably only interested in the more common two-parameter Weibull distribution because the actual location is zero. `fit` *can* accommodate this by specifing that the location is fixed parameter.
+
+```{code-cell} ipython3
+# params = stats.weibull_min.fit(data, loc=0)  # careful! this provides loc=0 as a *guess*
+params = stats.weibull_min.fit(data, floc=0)
+params
+```
+
+```{code-cell} ipython3
+stats.weibull_min.nnlf(params, data)
+```
+
+By relying on a `fit` method to "just work", users are encouraged to neglect some of these important details. 
+
+Instead, we suggest that it is almost as simple to fit distributions to data using the generic optimization tools provided by SciPy. However, this allows the user to perform the fit according to their preferred objective function, provides options when things don't work as expected, and helps the user to avoid the most common pitfalls.
+
+For instance, suppose our desired objective function is indeed the negative log likelihood function. It is simple to define as a function of the shape parameter `c` and scale only, leaving the location at its default of zero.
+
+```{code-cell} ipython3
+def nllf(params):
+    c, scale = params
+    X = Weibull(c=c) * scale
+    return -X.logpdf(data).sum()
+
+c_, loc_, scale_ = params
+nllf((c_, scale_))
+```
+
+To perform the minimization, we can use `scipy.optimize.minimize`.
+
+```{code-cell} ipython3
+from scipy import optimize
+eps = 1e-10  # numerical tolerance to avoid invalid parameters
+lb = [eps, eps]  # lower bound on `c` and `scale`
+ub = [10, 10]  # upper bound on `c` and `scale`
+x0 = [1, 1]  # guess to get optimization started
+bounds = optimize.Bounds(lb, ub)
+res_mle = optimize.minimize(nllf, x0, bounds=bounds)
+res_mle
+```
+
+The value of the objective function is essentially identical, and the PDFs are indistinguishable.
+
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+x = np.linspace(eps, 15, 300)
+
+c, scale = res_mle.x
+X = Weibull(c=c)*scale
+plt.plot(x, X.pdf(x), '-', label='numerical optimization')
+
+c, loc, scale = params
+Y = Weibull(c=c)*scale
+plt.plot(x, Y.pdf(x), '--', label='`scipy.stats.fit`')
+
+plt.hist(data, bins=np.linspace(0, 20, 30), density=True, alpha=0.1)
+plt.xlabel('x')
+plt.ylabel('pdf(x)')
+plt.legend()
+plt.ylim(0, 0.5)
+plt.show()
+```
+
+However, with this approach, it is trivial to make changes to the fitting procedure to suit our needs, enabling estimation procedures other than those provided by [`rv_continuous.fit`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_continuous.fit.html#scipy.stats.rv_continuous.fit). For instance, we can perform [maximum spacing estimation](https://en.wikipedia.org/wiki/Maximum_spacing_estimation) (MSE) by changing the objective function as follows.
+
+```{code-cell} ipython3
+def nlps(params):
```

Comment:
Could you add a comment what acronym `nlps` stands for?

Reply:
Sure. In the meantime, "negative log product spacing".